roles/setup-services/tasks/main.yml

```
@@ -30,59 +26,68 @@
     dest: /tmp/mobileclient.json
 
 - name: Create mobileclient
-  shell: oc apply -f /tmp/mobileclient.json -n {{ namespace }}
+  shell: oc apply -f /tmp/mobileclient.json -n {{ mobile_services_namespace }}
+
+- name: Remove temporary file
+  file:
+    path: /tmp/mobileclient.json
+    state: absent
+
+- name: Get uid of the mobileclient
+  shell: oc get mobileclient app1 -o jsonpath={.metadata.uid} -n {{ mobile_services_namespace }}
+  register: mobileclient_uid_output
+
+- set_fact: appuid={{ mobileclient_uid_output.stdout }}
+
+- name: Get route of the sync server
+  shell: oc get route ionic-showcase-server -o jsonpath={.spec.host} -n {{ namespace }}
+  register: sync_server_route_output
+
+- set_fact: sync_server_url={{ sync_server_route_output.stdout }}
 
 - include_tasks: create-binding.yml
   with_items:
     - binding-idm-app
-    - binding-idm-showcase-server
-    - binding-metrics-app
+    - binding-security
     - binding-sync-app
 
-- set_fact: bindings=4
-
 - name: Bind Push Android
   block:
+    - include_tasks: create-push-app.yml
     - include_tasks: create-binding.yml
       with_items:
         - binding-ups-app-android
-    - set_fact: bindings={{ bindings | int + 1}}
   when: push_android_server_key != "" and push_android_sender_id != ""
 
 - name: Bind Push iOS
   block:
+    - include_tasks: create-push-app.yml
     - include_tasks: create-binding.yml
       with_items:
         - binding-ups-app-ios
-    - set_fact: bindings={{ bindings | int + 1}}
   when: push_ios_cert_content != "" and push_ios_cert_passphase != "" and push_ios_cert_production != ""
 
-- name: Wait for all bindings to be completed
-  shell: oc get servicebindings -o=jsonpath='{.items[*].status.conditions[0].status}' -n {{namespace}} | tr ' ' '\n' | grep True| wc -w
-  register: bindings_ready_status_output
-  until: bindings_ready_status_output.stdout.find(bindings) != -1
+- name: Wait for Keycloak to be ready
+  shell: oc get keycloakrealm {{ appname }} -o jsonpath={.status.phase} -n {{ mobile_services_namespace }}
+  register: keycloak_status_output
+  until: keycloak_status_output.stdout == 'reconcile'
   retries: 50
   delay: 5
-  failed_when: false
-  changed_when: false
+
+- name: Assign roles to users
+  include_tasks: configure-keycloak.yml
+
+- name: Create keycloak config secret
+  shell: oc get --export secret {{ appname }}-server-install-config -o json -n {{ mobile_services_namespace }} | oc create -f - -n {{ namespace }}
```

```suggestion
  shell: oc get --export secret {{ appname }}-server-install-config -o json -n {{ mobile_services_namespace }} | oc apply -f - -n {{ namespace }}
```
`oc apply` makes sure running the playbook again won't fail